GeosCore/cldj_interface_mod.F90

```diff
@@ -0,0 +1,1200 @@
+             T0 = T0 + TREF2(K)*XC
+          ENDIF
+       ENDDO
+       T_CLIM(L)  = T0
+       O3_CLIM(L) = F0 * 1.e-6_fp
+    ENDDO
+
+    !=================================================================
+    ! Calculate effective altitudes using scale height at each level
+    !=================================================================
+    Z_CLIM(1) = 0.e+0_fp
+    DO L = 1, L_
+       SCALEH = BOLTZ * 1.e+4_fp * MASFAC * T_CLIM(L)
+       Z_CLIM(L+1) = Z_CLIM(L) - ( LOG( P_CTM(L+1) / P_CTM(L) ) * SCALEH )
+    ENDDO
+    Z_CLIM(L1_+1)=Z_CLIM(L1_) + ZZHT
+
+    !=================================================================
+    ! Calculate column quantities for Cloud-J
+    !=================================================================
+    PROFCOL = 0e+0_fp
+
+    DO L = 1, L1_
+
+       ! Monthly mean air Column [molec/cm2]
+       AIR_CLIM(L)  = ( P_CTM(L) - P_CTM(L+1) ) * MASFAC
+
+       ! Monthly mean O3 column [molec/cm2]
+       O3_CLIM(L) = O3_CLIM(L) * AIR_CLIM(L)
+
+       ! Monthly mean O3 column [DU]
+       PROFCOL = PROFCOL + ( O3_CLIM(L) / 2.69e+16_fp )
+    ENDDO
+
+!ewl: is this needed?
+    !! Top values are special (do not exist in CTM data)
+    !AIR_CLIM(L1_)     = P_CTM(L1_) * MASFAC
+    !O3_CLIM(L1_) = O3_CLIM(L1_) * AIR_CLIM(L1_)
+
+    ! Scale monthly O3 profile to the daily O3 profile (if available)
+    DO L = 1, L1_
+
+       ! Use online O3 values in the chemistry grid if selected; otherwise use
+       ! O3 values from the met fields or TOMS/SBUV
+       IF ( ( Input_opt%Use_Online_O3 )             &
+            .AND. ( L <= State_Grid%MaxChemLev )    &
+            .AND. ( O3_CTM(L) > 0e+0_fp ) ) THEN
+
+          ! Convert from molec/cm3 to molec/cm2
+          O3_CLIM(L) = O3_CTM(L) * (Z_CLIM(L+1)-Z_CLIM(L))
+
+       ELSEIF (State_Chm%TO3_Daily(ILON,ILAT) > 0e+0_fp) THEN
+
+          ! NOTE: replaced O3_TOMS with State_Chm%TO3_Daily since is the equivalent
+          O3_CLIM(L) = O3_CLIM(L) * ( State_Chm%TO3_Daily(ILON,ILAT) / PROFCOL )
+
+       ENDIF
+
+    ENDDO
+
+  END SUBROUTINE Set_Clim_Profiles
+!EOC
+!------------------------------------------------------------------------------
+!                  GEOS-Chem Global Chemical Transport Model                  !
+!------------------------------------------------------------------------------
+!BOP
+!
+! !IROUTINE: Get_RH_Index
+!
+! !DESCRIPTION: Function GET\_RH_\Index gets index corresponding to relative
+!  humidity bin used in Cloud-J tables (1-5).
+!\\
+!\\
+! !INTERFACE:
+!
+  FUNCTION Get_RH_Index( RH ) RESULT( index )
+!
+! !INPUT PARAMETERS:
+!
+    REAL(fp), INTENT(IN) :: RH  ! Relative humidity [%]
+!
+! !RETURN VALUE:
+!
+    INTEGER :: index
+!
+! !REMARKS:
+!
+! !REVISION HISTORY:
+!  29 Nov 2023 - R. Yantosca - Initial version
+!  See https://github.com/geoschem/geos-chem for complete history
+!EOP
+!------------------------------------------------------------------------------
+!BOC
+
+    IF ( RH <= 50 ) index = 1
```

Comment:
Oops @lizziel, I think I messed this up.  The RETURN statements should be in the IF blocks, i.e.
```F90
IF ( RH <= 50 ) THEN
   index = 1
   RETURN
ENDIF
```
Otherwise it'll just return after the first IF block.  My bad.